cirq-core/cirq/experiments/two_qubit_xeb.py

```diff
@@ -0,0 +1,178 @@
+    import cirq
+
+
+def _grid_qubits_for_sampler(sampler: 'cirq.Sampler'):
+    if hasattr(sampler, 'processor'):
+        device = sampler.processor.get_device()
+        return sorted(device.metadata.qubit_set)
+    else:
+        qubits = devices.GridQubit.rect(3, 2, 4, 3)
+        # Delete one qubit from the rectangular arangement to
+        # 1) make it irregular 2) simplify simulation.
+        return qubits[:-1]
+
+
+def _manhattan_distance(qubit1: cirq.GridQubit, qubit2: cirq.GridQubit) -> int:
+    return abs(qubit1.row - qubit2.row) + abs(qubit1.col - qubit2.col)
+
+
+class TwoQubitRandomizedBenchMarkResult:
+    def __init__(self, fidelities: pd.DataFrame) -> None:
+        self.fidelities = fidelities
+        self._qubit_pair_map = {idx[-1]: i for i, idx in enumerate(fidelities.index)}
+
+    def plot_heatmap(self, ax: Optional[plt.Axes] = None, **plot_kwargs):
+        show_plot = not ax
+        if not ax:
+            fig, ax = plt.subplots(1, 1, figsize=(8, 8))
+
+        heatmap_data = {pair: self.depolarization_error(*pair) for pair in self.all_qubit_pairs}
+
+        ax.title('device depolarization error heatmap')
+        vis.TwoQubitInteractionHeatmap(heatmap_data).plot(ax=ax, **plot_kwargs)
+        if show_plot:
+            fig.show()
+
+    def plot_fitted_exponential(
+        self,
+        q0: 'cirq.GridQubit',
+        q1: 'cirq.GridQubit',
+        ax: Optional[plt.Axes] = None,
+        **plot_kwargs,
+    ):
+        show_plot = not ax
+        if not ax:
+            fig, ax = plt.subplots(1, 1, figsize=(8, 8))
+        if q0 > q1:
+            q0, q1 = q1, q0
+        record = self.fidelities.iloc[self._qubit_pair_map[(q0, q1)]]
+
+        plt.axhline(1, color='grey', ls='--')
+        plt.plot(record['cycle_depths'], record['fidelities'], 'o')
+        xx = np.linspace(0, np.max(record['cycle_depths']))
+
+        ax.plot(
+            xx,
+            exponential_decay(xx, a=record['a'], layer_fid=record['layer_fid']),
+            label='estimated exponential decay',
+            **plot_kwargs,
+        )
+        ax.title(f'{q0}-{q1}')
+        ax.ylabel('Circuit fidelity')
+        ax.xlabel('Cycle Depth $d$')
+        ax.legend(loc='best')
+        if show_plot:
+            fig.show()
+
+    def depolarization_error(self, q0: 'cirq.GridQubit', q1: 'cirq.GridQubit'):
+        if q0 > q1:
+            q0, q1 = q1, q0
+        p = self.fidelities.layer_fid[self._qubit_pair_map[(q0, q1)]]
+        return 1 - p
+
+    def pauli_error(self, q0: 'cirq.GridQubit', q1: 'cirq.GridQubit'):
+        return self.depolarization_error(q0, q1) * (1 - 1 / 16)
+
+    def average_error(self, q0: 'cirq.GridQubit', q1: 'cirq.GridQubit'):
+        return self.depolarization_error() * (1 - 1 / 4)
+
+    @functools.cached_property
+    def all_qubit_pairs(self) -> frozenset[Tuple['cirq.GridQubit', 'Cirq.GridQubit']]:
+        return frozenset(self._qubit_pair_map.keys())
+
+
+def parallel_two_qubit_randomized_benchmarking(
```

Comment:
This is XEB, not randomized benchmarking, correct?

Reply:
fixed